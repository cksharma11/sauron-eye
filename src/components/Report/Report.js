import React from "react";
import FailedTests from "../FailedTests/FailedTests";

const Report = ({ detail, username }) => {
  const { time, sha, failedTests } = detail;
  return (
    <div className="report">
      <p>{`${username} | ${sha} | ${time}`}</p>
      <FailedTests failedTests={failedTests} />
    </div>
  );
};

export default Report;
