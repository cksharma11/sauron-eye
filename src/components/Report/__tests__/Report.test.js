import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Report from "../Report";

configure({ adapter: new Adapter() });

describe("Report", () => {
  it("should render properly", () => {
    const detail = {
      sha: "sha123",
      time: new Date(),
      failedSuites: []
    };
    const username = "test";
    const wrapper = shallow(<Report detail={detail} username={username} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
