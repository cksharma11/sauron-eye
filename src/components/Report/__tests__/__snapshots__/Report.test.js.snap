// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report should render properly 1`] = `
"<div className=\\"report\\">
  <p>
    test | sha123 | 0 seconds ago
  </p>
  <FailedTests failedSuites={{...}} />
</div>"
`;
